install-python-poetry/action.yml

```diff
@@ -0,0 +1,92 @@
+name: Install Python Poetry
+description: This action installs and sets up Python Poetry
+
+inputs:
+  poetry_version:
+    type: string
+    description: The version of poetry that will be installed. Omitting this input will install the latest version of poetry.
+    required: false
+    default: ''
+  poetry_home:
+    type: string
+    description: Set a custom directory for POETRY_HOME
+    required: false
+    default: ''
+  add_poetry_to_path:
+    type: boolean
+    description: Update PATH so `poetry` can be invoked directly. Defaults to true
+    required: false
+    default: true
+
+
+outputs:
+  poetry_home:
+    description: 'Path to directory where poetry is installed'
+    value: ${{ steps.poetry.outputs.home }}
+  poetry_bin:
+    description: 'Path to poetry binary'
+    value: ${{ steps.poetry.outputs.bin }}
+
+
+runs:
+  using: composite
+  steps:
+    - name: Check Python available
+      shell: bash
+      run: |
+        if ! command -v python &> /dev/null
+        then
+          echo "::error title=Missing Python::Could not find a python installation. Make sure to use actions/setup-python prior to calling this action."
+          exit 1
+        fi
+
+    - name: Check Poetry Version Input
+      shell: bash
+      id: poetry_version
+      if: inputs.poetry_version != ''
+      env:
+        INPUT_POETRY_VERSION: ${{ inputs.poetry_version }}
+      run: echo "version===$INPUT_POETRY_VERSION" >> $GITHUB_OUTPUT
```

Comment:
Probably better in terms of readability: `version=$INPUT_POETRY_VERSION` and then at line 72 `pip install poetry==${{ steps.poetry_version.outputs.version }}`. More complex with '===' probably.

Reply:
Issue is, if I do that, and the calling workflow does not pass a version (as in, they want to install the latest available), then line 72 would end up with `poetry==` and that would raise an error.

The way it is right now, the `==<version>` is only tacked on if a version is actually specified, otherwise line 72 simply renders as `pip install poetry`

Reply:
I see, you're right. Didn't consider that specific case, then that's fine as it is! 👍

Reply:
NP! I'll add a comment to this block to describe this behaviour as others might ponder upon this same question in the future

Reply:
Comment added!